libcraft/items/src/inventory_slot.rs

```diff
@@ -99,31 +98,31 @@ impl InventorySlot {
     }
 
     /// Returns the number of items stored in the inventory slot.
+    #[must_use]
     pub fn count(&self) -> u32 {
-        match self {
-            InventorySlot::Filled(stack) => stack.count(),
-            InventorySlot::Empty => 0,
-        }
+        self.map_ref(ItemStack::count).unwrap_or(0)
     }
 
     /// Should only be called if the caller can guarantee that there is space
-    /// such that the new could is not greater then self.stack_size().
+    /// such that the new could is not greater then `self.stack_size`().
     /// And that the slot actually contains an item.   
     fn add_count(&mut self, n: u32) {
-        match self {
-            InventorySlot::Filled(x) => x.add(n).unwrap(),
-            InventorySlot::Empty => panic!("add count called on empty inventory slot!"),
-        };
+        self.option_mut()
+            .expect("add count called on empty inventory slot!")
```

Comment:
I don't like this panic. Adding to count of an empty slot is still an empty slot?

Reply:
It is an internal function, so I think aborting on any logic error is ok.